config/config.example.json

```diff
@@ -8,7 +8,8 @@
     "userId": {
       "id": "example_userid",
       "type": "example_userid_type"
-    }
+    },
+    "implementation": "icare"
```

Comment:
DM2: I think this might be more a question for @Dtphelan1 but since we already default the `implementation` to `'icare'` for our operations on the ICARECSVClient, do we actually want this in the config example? I guess this is the repo for the base *ICARE* extraction client so it might make sense for that to be there, but I guess I was thinking of the more generic case where we make other clients and might not want to flag the implementation field explicitly, since we'll already default it to what it should be on a case-by-case basis.

Reply:
This actually gets to comments I'm making in the other repo; if we have a default implementation, it should be `mcode` not `icare`. Fixing that will make it clear why we want a configuration variable here. What's more, we could probably make use of a switch statement and avoid having to provide a default argument value. This would also make es-lint happy and avoid a gloabl disable of an eslint rule that we have there right now. 

Let me know if I missed anything.

Reply:
DM2: I think this makes a lot of sense. I'm partial to keeping the `implementation` flag in the example config but maybe adding a line communicating that the flag is optional, and will default to `"icare"` otherwise. I feel like if the flag is supported in the config file then it should be noted in the example, but I don't feel too strongly either way so I'll go with the consensus.

Reply:
Further on DM2: Thanks both for the answers! I think I still stand by the assessment that maybe we don't want it included in the example config. There's not really a way in a JSON file to safely include a comment, so I don't think we can annotate it like you suggested, Julian. Otherwise I'd agree that would be the preferable solution. Given that, if both of you agree, I think I'd rather keep it out until such point as we have a reason to include it when we have other implementations to manage.

Reply:
Julian I think you bring up a strong point in general, which is that our documentation for arguments in our config is terrible right now. A hybrid approach could be to remove this flag from our config now as per @kjmahalingam request, but to create a task in the backlog to "improve our documentation on recognized flags/variables in our configuration file". This might factor into the packaging tasks that we have in the sprint, or fall out into a separate task all together. Thoughts?

Reply:
DM2: Just seeing the reply from @Dtphelan1. That makes sense! Especially given that we don't expect the `implementation` flag to be changed, or even used all that often. I'll go ahead and remove the flag from the example config.